numba/np/npyfuncs.py

```diff
@@ -1590,23 +1638,39 @@ def np_real_nextafter_impl(context, builder, sig, args):
     _check_arity_and_homogeneity(sig, args, 2)
 
     dispatch_table = {
-        types.float32: 'npy_nextafterf',
-        types.float64: 'npy_nextafter',
+        types.float32: 'numba_nextafterf',
+        types.float64: 'numba_nextafter',
     }
 
     return _dispatch_func_by_name_type(context, builder, sig, args,
                                        dispatch_table, 'nextafter')
 
 def np_real_spacing_impl(context, builder, sig, args):
+    # This is different to how NumPy does it, NumPy has a specialisation of
+    # nextafter called _next, which is used. See:
+    # https://github.com/numpy/numpy/blob/12c2b7dd62fc0c14b81c8892ed5f4f59cc94d09c/numpy/core/src/npymath/ieee754.c.src#L32-L38
+    # Numba elects to use `nextafter` for a similar behaviour to save
+    # translating this very involved function. Further, the NumPy comments note
+    # that there is a lot of redundancy present between the two.
     _check_arity_and_homogeneity(sig, args, 1)
 
     dispatch_table = {
-        types.float32: 'npy_spacingf',
-        types.float64: 'npy_spacing',
+        types.float32: 'numba_nextafterf',
+        types.float64: 'numba_nextafter',
     }
 
```

Comment:
`*sig.args, *sig.args` is an odd way to spell this. 

wouldn't something like below be more readable

```python
[ty] = sig.args
typing.signature(sig.return_type, ty, ty)
```

Reply:
Fixed in 7966e9b.